src/components/SingleMovie.js

```diff
@@ -1,33 +1,57 @@
 import React from "react"
 import "../css/SingleMovie.css"
+import getData from "../apiCalls"
 
-function SingleMovie ({movieInFocus, movies}) {
 
-  const correctMovie = movies.find(movie => movie.id === movieInFocus)
+class SingleMovie extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      id:props.movieInFocus,
+      correctMovie:{},
+      error:""
+    }
+  }
 
-  return (
-    <div className="SingleMovie" style={{
-      backgroundImage:`linear-gradient(to bottom, transparent 0%, black 60%),
-      url(${correctMovie.backdrop_path})`,
-      backgroundPosition: 'center',
-      backgroundSize: 'cover',
-      backgroundRepeat: 'no-repeat',
-      width:'100vw',
-      height:'100vh'
-      }}>
-      <h3 className="single-movie-title"> 
-        {correctMovie.title}
-        <h4 className="single-movie-tagline"> {correctMovie.tagline}</h4>
-      </h3>
-      <div className="single-movie-info-container">
-        <p className="single-movie-overview">Overview:<br></br>{correctMovie.overview}</p>
-        <p>Average User Rating: {Math.round(correctMovie.average_rating)}</p>
-        <p className="single-movie-info">
-          Release Date: {correctMovie.release_date} Genre:{correctMovie.genres} Revenue: ${correctMovie.revenue} Budget: ${correctMovie.budget}
-        </p>
-      </div>
-    </div>
-  )
+  componentDidMount = () => {
+    getData(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${this.state.id}`)
+      .then(data => {this.setState({correctMovie: data.movie})})
+      .catch(error => {
+        this.setState({error:error.message})
+      })
+    }
+
+  render() {
+    
+    const { correctMovie:{title, backdrop_path, tagline, overview, average_rating, release_date, genres, revenue, budget}, error } = this.state;
+
+    return error ? <p className="error">Uh oh! Something went wrong</p>: 
+      <div className="SingleMovie" style={{
+          backgroundImage:`linear-gradient(to bottom, transparent 0%, black 60%),
+          url(${backdrop_path})`,
+          backgroundPosition: 'center',
+          backgroundSize: 'cover',
+          backgroundRepeat: 'no-repeat',
+          width:'100vw',
+          height:'100vh'
+          }}>
+          <h3 className="single-movie-title"> 
+            {title}
+            <p className="single-movie-tagline"> {tagline}</p>
+          </h3>
+          <div className="single-movie-info-container">
+            <p className="single-movie-overview">Overview:<br></br>{overview}</p>
+            <p>Average User Rating: {Math.round(average_rating)}</p>
+            <div className="single-movie-info">
+              <p>Release Date: {release_date}</p>
+              {!!genres && <p>Genre: {genres.join(", ")}</p>} 
```

Comment:
This is such genius use of !!, took me a second to understand how it was working, but this is such rad conditional rendering. Awesome job!